pkg/logql/log/filter.go

```diff
@@ -112,18 +161,21 @@ func (a andFilter) ToStage() Stage {
 	}
 }
 
+func (a andFilter) Matches(test Checker) bool {
+	return a.left.Matches(test) && a.right.Matches(test)
+}
+
 type andFilters struct {
-	filters []Filterer
+	filters []MatcherFilterer
 }
 
-// NewAndFilters creates a new filter which matches only if all filters match
-func NewAndFilters(filters []Filterer) Filterer {
+func NewAndMatcherFilters(filters []MatcherFilterer) MatcherFilterer {
```

Comment:
Is this anywhere used except in `NewAndFilters`, which returns `Filterer`?

Reply:
Nope, removed